src/caret_analyze/record/records_service/latency.py

```diff
@@ -81,15 +83,21 @@ def to_records(
         records = self._create_empty_records()
 
         for start_ts, end_ts in zip(self._start_timestamps, self._end_timestamps):
+            latency = end_ts - start_ts
+            if latency < 0:
+                warn('Record data is invalid. '
+                     'The end time of the path is recorded before the start time.',
+                     UserWarning)
+                continue
             if converter:
                 record = {
                     self._start_column: round(converter.convert(start_ts)),
-                    'latency': end_ts - start_ts
+                    'latency': max(end_ts - start_ts, 0)
                 }
             else:
                 record = {
                     self._start_column: start_ts,
-                    'latency': end_ts - start_ts
+                    'latency': max(end_ts - start_ts, 0)
```

Comment:
I suggest using variable defined above.

```suggestion
                    'latency': latency
```

Reply:
Fixed in [cc43ec7](https://github.com/tier4/caret_analyze/pull/455/commits/cc43ec754a8a2f0cc4fc2f14d9479bde3042e19b).